xarray/backends/common.py

```diff
@@ -96,8 +98,9 @@ def load(self):
         This function will be called anytime variables or attributes
         are requested, so care should be taken to make sure its fast.
         """
-        variables = FrozenOrderedDict((_decode_variable_name(k), v)
-                                      for k, v in iteritems(self.get_variables()))
+        self.encoding = self.get_encoding()
```

Comment:
This is a little dangerous -- `.load()` needs to be called in order to guarantee a consistent `encoding`  state on a DataStore. I would rather we didn't set such state, and simply pulled this information out of the file linked to the DataStore as necessary.